### PR TITLE
add bintray publish task

### PR DESCRIPTION
**Pull request by komatatsu**
add task `publish` fbba73d
published v2.1.0
https://bintray.com/cookpad-inc/maven/issue-reporter

and add task `releng`   868deb5
releng task is set git tag

---
# how to publish for jcenter
1. create [bintray](https://bintray.com/) account
2. get API_KEY from your profile edit page
3. `./gradlew clean build bintrayUpload -PbintrayUser=[your account name] -PbintrayKey=[your api key] -PdryRun=true`
4. change to `-PdryRun=false`
5. `./gradlew releng`

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Thu Dec 11 10:31:44 JST 2014
+#Tue Sep 06 11:03:59 JST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.2.1-all.zip
```
Comment (nein37) [inline gradle/wrapper/gradle-wrapper.properties]: バージョン下がってる？

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.2-all.zip
```
Comment (hogelog) [inline gradle/wrapper/gradle-wrapper.properties]: downgrade?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (komatatsu) [inline gradle/wrapper/gradle-wrapper.properties]: fix 7fcf88f

File: README.md [hunk 3]
```diff
@@ -39,3 +39,13 @@ protected void onCreate(Bundle savedInstanceState) {
...
 ```
+
+# For Developers
+
+## How to publish for jcenter
+
+1. create [bintray](https://bintray.com/) account
+2. get API_KEY from your profile edit page
+3. `./gradlew clean build bintrayUpload -PbintrayUser=[your account name] -PbintrayKey=[your api key] -PdryRun=true`
+4. change to `-PdryRun=false`
```
Comment (kazu-matsu) [inline README.md]: Does this mean we should run with `./gradlew clean build bintrayUpload -PbintrayUser=[your account name] -PbintrayKey=[your api key] -PdryRun=false` ?

**Comment by nein37** [comment]
LGTM

File: README.md (hunk 3 above)
Comment (komatatsu) [inline README.md]: maybe, out some error

Comment (komatatsu) [inline README.md]: no, I fixed error.
don't need dryrun

File: library/build.gradle [hunk 4]
```diff
@@ -1,4 +1,6 @@
 apply plugin: 'com.android.library'
+apply plugin: 'com.novoda.bintray-release'
+apply plugin: 'me.tatarka.retrolambda'
```
Comment (tomorrowkey) [inline library/build.gradle]: it could be not necessary.

File: README.md (hunk 3 above)
Comment (komatatsu) [inline README.md]: 86fbb5d

File: README.md [hunk 5]
```diff
@@ -46,6 +46,5 @@ protected void onCreate(Bundle savedInstanceState) {
 
 1. create [bintray](https://bintray.com/) account
 2. get API_KEY from your profile edit page
-3. `./gradlew clean build bintrayUpload -PbintrayUser=[your account name] -PbintrayKey=[your api key] -PdryRun=true`
-4. change to `-PdryRun=false`
-5. `./gradlew releng`
+3. `./gradlew clean build bintrayUpload -PbintrayUser=[your account name] -PbintrayKey=[your api key] -PdryRun=false`
+4. `./gradlew releng`
```
Comment (kazu-matsu) [inline README.md]: pretty good 👍

File: library/build.gradle (hunk 4 above)
Comment (komatatsu) [inline library/build.gradle]: fix bce8551

File: library/build.gradle [hunk 6]
```diff
@@ -28,9 +29,47 @@ android {
...
+        throw new GradleException(proc.err.text.trim())
+    }
+    return proc.in.text;
+}
+
+task 'releng' << {
+    def timestamp = new Date().format("yyyyMMddHHmm")
+    def tag = "v${android.defaultConfig.versionName}-${timestamp}"
+
+    println "[releng] makeing a release engineering for $tag (${git('rev-parse HEAD').trim()}))"
```
Comment (hotchemi) [inline library/build.gradle]: making?

Comment (komatatsu) [inline library/build.gradle]: 28828b0

File: library/build.gradle [hunk 7]
```diff
@@ -28,9 +29,47 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_7
         targetCompatibility JavaVersion.VERSION_1_7
     }
+
+    android {
+        lintOptions {
+            abortOnError false
```
Comment (tomorrowkey) [inline library/build.gradle]: [IMO] Why ignore lint errors? If the errors detected, that should be fixed.

Comment (komatatsu) [inline library/build.gradle]: can't publish if there is a Lint error

**Comment by tomorrowkey** [comment]
LGTM without comments
Thanks for maintenance ☺️

**Pull request merged**